uber/site_sections/preregistration.py

```diff
@@ -524,3 +527,22 @@ def shirt_reorder(self, session, message='', **params):
             'message': message,
             'attendee': attendee
         }
+
+    def apply_promo_code(self, session, message='', **params):
+        if 'id' and 'code' in params:
+            if params['code'] is not '':
+                attendee, group = self._get_unsaved(params['id'])
+                if not attendee:
```

Comment:
I couldn't figure out how to apply the discount to a group in the Charge statement, so I decided to build purely for single attendee first.